feat: Helm installation for integrated exmaple

TrevorLink opened the pull request:
### Describe what this PR does / why we need it
Helm chart templates for integrated-example deployment on kubernetes


### Does this pull request fix one issue?

#2697

TrevorLink committed:
feat: Helm installation for integrated example on kubernetes

TrevorLink committed:
doc: add Helm installation docs for integrated example

TrevorLink committed:
polish yaml for kubernetes resource

TrevorLink committed:
doc: remove useless statements in docs

steverao commented on spring-cloud-alibaba-examples/integrated-example/docs/zh/kubernetes-deployment-zh.md:
```diff
@@ -1,2 +1,113 @@
+
+## 准备工作
+
+此版本为 Spring Cloud Alibaba 最佳实践 Kubernetes 部署版本，其中需要您准备如下的环境
```
Add ':' in line tail, please pay attention to punctuation in your document.

steverao commented on spring-cloud-alibaba-examples/integrated-example/docs/en/kubernetes-deployment.md:
```diff
@@ -1,2 +1,115 @@
+##### start test
+
+
+Visit `http://127.0.0.1:30080/order` to experience the corresponding scenario.
```
It shouldn't use 127.0.0.1 in K8s. there are some same problems below.

steverao commented on spring-cloud-alibaba-examples/integrated-example/docs/zh/kubernetes-deployment-zh.md:
```diff
@@ -1,2 +1,113 @@
+
+## 启动测试
+
+进入到 `spring-cloud-alibaba-examples/integrated-example` 目录下，执行如下命令完成 Helm 的安装
```
这句话说法有点问题，不是完成Helm的安装，是利用Helm部署应用程序。

TrevorLink committed:
make local version code same as kubernetes version code

TrevorLink committed:
add dockerfile for each microservice

TrevorLink committed:
add public image repo

TrevorLink committed:
doc: polish doc with syntax mistake

doc: polish doc format

doc: polish kubernetes doc

doc: polish local version doc

doc: polish k8s doc

steverao commented on spring-cloud-alibaba-examples/integrated-example/docs/zh/kubernetes-deployment-zh.md:
````diff
@@ -1,2 +1,121 @@
+此外，在启动测试前还需配置好 Kubernetes 主节点的 host 配置文件映射。
+
+```sh
+127.0.0.1 integrated-frontend
````
You should list all NodePort name here, otherwise, we request the specific NodePort service that will failed. By the way, I found there isn't a NodePort name called `integrated-frontend`.

steverao commented on spring-cloud-alibaba-examples/integrated-example/scripts/nacos-config-quick.sh:
```diff
@@ -8,11 +7,12 @@ orderConfig=$(cat ../config/integrated-order.yaml)
-curl -X POST "127.0.0.1:8848/nacos/v1/cs/configs" -d "dataId=integrated-provider.yaml&group=${groupId}&content=${providerConfig}"
-curl -X POST "127.0.0.1:8848/nacos/v1/cs/configs" -d "dataId=integrated-consumer.yaml&group=${groupId}&content=${consumerConfig}"
+groupId="integrated-example"
+curl -X POST "nacos-server:8848/nacos/v1/cs/configs" -d "dataId=datasource-config.yaml&group=DEFAULT_GROUP&content=${datasourceConfig}"
```
Why don't use same groupId between below here? Is it a mistake?

steverao commented on spring-cloud-alibaba-examples/integrated-example/helm-chart/templates/integrated-nacos-mysql.yaml:
```diff
@@ -0,0 +1,57 @@
+    spec:
+      containers:
+        - name: mysql
+          image: trevorlink/nacos-mysql
```
There is a mistake here that needs to be corrected. BTW, I didn't see the pod when I deployed the whole demo and please check whether it is a valuable content?

steverao commented on spring-cloud-alibaba-examples/integrated-example/helm-chart/templates/integrated-rocketmq.yaml:
```diff
@@ -0,0 +1,65 @@
+    spec:
+      containers:
+        - name: mqnamesrv
+          image: liuyi71sinacom/rocketmq-4.8.0
```
You can download the image and push our namespace of sca-community.

steverao commented on spring-cloud-alibaba-examples/integrated-example/helm-chart/templates/integrated-rocketmq.yaml:
```diff
@@ -0,0 +1,65 @@
+    spec:
+      containers:
+        - name: mqbroker
+          image: liuyi71sinacom/rocketmq-4.8.0
```
Same problem like above.

steverao commented on spring-cloud-alibaba-examples/integrated-example/docs/zh/local-deployment-zh.md:
````diff
@@ -20,9 +20,22 @@
+为了保证代码可以正常启动，请先配置好本机的 host 映射，在配置文件中新增如下的映射。
+```sh
+# for integrated-example
+127.0.0.1 integrated-mysql
````
Is the quantity complete? Don't need to add all NodePort and ClusterIP of K8s as host here?

TrevorLink committed:
keep config and register group all the same as 'integrated-example'

TrevorLink committed:
modify image prefix to sca-community

TrevorLink committed:
doc: add node port and cluster ip export docs

steverao commented on spring-cloud-alibaba-examples/integrated-example/docs/en/kubernetes-deployment.md:
```diff
@@ -1,2 +1,123 @@
+- Sentinel binds specified gateway routes on the gateway side for fusion degradation of services.
+- RocketMQ performs traffic clipping, where the producer sends messages to RocketMQ under high traffic requests, while the consumer pulls and consumes through a configurable consumption rate, reducing the pressure of high traffic direct requests to the database to increase the number of likes requests.
+
+#### startup test
```
Please note the capitalization of the first letter.

steverao commented on spring-cloud-alibaba-examples/integrated-example/docs/en/kubernetes-deployment.md:
```diff
@@ -1,2 +1,123 @@
+- Deduct the account balance
+- Generate order information to return a response
+
+##### start test
```
Please note the capitalization of the first letter and check whether there are same problems in your PR.

steverao commented on spring-cloud-alibaba-examples/integrated-example/helm-chart/templates/integrated-frontend.yaml:
```diff
@@ -0,0 +1,47 @@
+          ports:
+            - name: http-port
+              containerPort: 8080
+      affinity:
```
The below content is useless, because you don't have specific rule.

TrevorLink committed:
doc: polish doc for some format and capitalization letter problem

TrevorLink committed:
remove useless declaration in kubernetes resource yaml

TrevorLink committed:
remove useless declaration in k8s resource for business module

TrevorLink committed:
doc: add kubernetes node port & cluster ip config

steverao commented:
LGTM.
